### PR TITLE
Add proptypes guidelines

### PR DESCRIPTION
**Pull request by manjumuthaiya**

### CODE REVIEW

File: react/README.md [hunk 1]
```diff
@@ -297,6 +298,72 @@
...
+      AppContainer.propTypes = {
+        status: statusPropTypes,
+        data: PropTypes.string.isRequired,
+      };
+    ```
+  - When a subset of a component's proptypes can be used in multiple components, export it as a `const` from the component file.
+
+    ```
+      export const partialFilterPropTypes = {
+        searchString: PropTypes.string,
```
Comment (sfletche) [inline react/README.md]: might add a second prop here (or maybe just a `...`) to make it clear that we wouldn't do this unless there were multiple props

Comment (manjumuthaiya) [reply react/README.md]: done 👍

File: react/README.md [hunk 2]
```diff
@@ -297,6 +298,72 @@
...
+      };
+      FilterData.propTypes = {
+        ...partialFilterPropTypes,
+      };
+    ```
+  - When a set of proptypes are not associated with a single component, but can be used across multiple components, define them in a `constants/proptypes.js` file. Example:
```
Comment (sfletche) [inline react/README.md]: is this specific to `shape`?  

if not, it's not entirely clear to me how to differentiate between the previous rule and this rule (when do we export the proptypes from a component, and when do we export it from `proptypes.js`?)

File: react/README.md [hunk 3]
```diff
@@ -297,6 +298,74 @@
...
+      export const partialFilterPropTypes = {
+        searchString: PropTypes.string,
+        onFilter: PropTypes.func,
+      };
+      FilterData.propTypes = {
+        ...partialFilterPropTypes,
+        isFilterActive: PropTypes.bool.isRequired,
+      };
+    ```
+  - When a set of proptypes are not associated with a single component, but can be used across multiple components, define them in a `constants/proptypes.js` file. Example:
```
Comment (sfletche) [inline react/README.md]: is this specific to `shape`?  

if not, it's not entirely clear to me how to differentiate between the previous rule and this rule (when do we export the proptypes from a component, and when do we export it from `proptypes.js`?)

Comment (manjumuthaiya) [reply react/README.md]: this is referring to data that could be used across champ. Like `application` , or maybe `programDefinition` or `userData` or `location`

Comment (manjumuthaiya) [reply react/README.md]: do you have any ideas on a better way to express that as a style guide rule ?

File: react/README.md [hunk 4]
```diff
@@ -297,6 +298,74 @@
...
+    ```
+      import StatusMessage, { statusPropTypes } from './status-message';
+
+      class AppContainer { };
+      AppContainer.propTypes = {
+        status: statusPropTypes,
+        data: PropTypes.string.isRequired,
+      };
+    ```
+  - When a subset of a component's proptypes can be used in multiple components, export it as a `const` from the component file.
```
Comment (sfletche) [inline react/README.md]: might specify 3 or more here
`in multiple (3 or more) components`

File: react/README.md [hunk 5]
```diff
@@ -297,6 +298,74 @@
...
+      export const partialFilterPropTypes = {
+        searchString: PropTypes.string,
+        onFilter: PropTypes.func,
+      };
+      FilterData.propTypes = {
+        ...partialFilterPropTypes,
+        isFilterActive: PropTypes.bool.isRequired,
+      };
+    ```
+  - When describing a shape that is associated with multiple components, define them in a `constants/proptypes.js` file. Example:
```
Comment (sfletche) [inline react/README.md]: might adjust this slightly to 
`define the shape in a ....`

**Comment by sfletche** [comment]
👍

**Pull request merged**